functions/function/http-request-function/src/main/java/org/springframework/cloud/fn/http/request/HttpRequestFunctionConfiguration.java

```diff
@@ -50,36 +53,65 @@ public WebClient webClient() {
 	}
 
 	@Bean
-	public HttpRequestFunction httpRequestFunction(WebClient webClient, HttpRequestProperties properties) {
+	public HttpRequestFunction httpRequestFunction(WebClient webClient, HttpRequestFunctionProperties properties) {
 		return new HttpRequestFunction(webClient, properties);
 	}
 
 	/**
 	 * Function that accepts a {@code Flux<Message<?>>} containing body and headers and
 	 * returns a {@code Flux<ResponseEntity<?>>}.
 	 */
-	public static class HttpRequestFunction implements Function<Flux<Message<?>>, Flux<ResponseEntity<?>>> {
+	public static class HttpRequestFunction implements Function<Flux<Message<?>>, Flux<?>> {
 		private final WebClient webClient;
 
 		private final UriBuilderFactory uriBuilderFactory = new DefaultUriBuilderFactory();
 
-		private final HttpRequestProperties properties;
+		private final HttpRequestFunctionProperties properties;
 
-		public HttpRequestFunction(WebClient webClient, HttpRequestProperties properties) {
+		public HttpRequestFunction(WebClient webClient, HttpRequestFunctionProperties properties) {
 			this.webClient = webClient;
 			this.properties = properties;
 		}
 
 		@Override
-		public Flux<ResponseEntity<?>> apply(Flux<Message<?>> messageFlux) {
+		public Flux<?> apply(Flux<Message<?>> messageFlux) {
 			return messageFlux.flatMap(message -> this.webClient
-					.method(properties.getHttpMethod())
-					.uri(uriBuilderFactory.uriString(properties.getUrl()).build())
-					.bodyValue(properties.getBody() == null ? message.getPayload() : properties.getBody())
-					.headers(httpHeaders -> httpHeaders.addAll(properties.getHeaders()))
+					.method(resolveHttpMethod(message))
+					.uri(uriBuilderFactory.uriString(resolveUrl(message)).build())
+					.bodyValue(resolveBody(message))
+					.headers(httpHeaders -> httpHeaders.addAll(resolveHeaders(message)))
 					.retrieve()
 					.toEntity(properties.getExpectedResponseType())
+					.map(responseEntity -> properties.getReplyExpression().getValue(responseEntity))
 					.timeout(Duration.ofMillis(properties.getTimeout())));
 		}
+
+		private String resolveUrl(Message<?> message) {
+			return properties.getUrlExpression().getValue(message, String.class);
```

Comment:
Looks like you don't use any `EvaluationContext`, so we don't have access to bean factory from target expressions.

Reply:
The spelConverter from common-config sets it. https://github.com/spring-cloud/stream-applications/blob/master/functions/common/config-common/src/main/java/org/springframework/cloud/fn/common/config/SpelExpressionConverterConfiguration.java#L77 Does that work?

Reply:
Yep! It does. Thank you!